### PR TITLE
feat: add new list feed and adjust feed layout conditions

### PR DESCRIPTION
**Pull request by nensidosari**
## Changes
- using useFeedLayout to check for experiment flag  and enrol users on condition of being in list mode
- made adjustments in many pages to accommodate the new cards in list mode
- fix source/tags page horizontal feed on list layout V1
- squad page no has a list mode

### Describe what this PR does
- Enabled the new Card to be available in list mode behind an experiment
- confirmation about experiment with product https://dailydotdev.slack.com/archives/C02JAUF8HJL/p1715679157241849

P.s current listMode or insaneMode has other conditions throughout the code, I'm at the moment considering only the setting coming from SettingsContext and seem to be working well, this might need quite a lot of cleanup if this experiment wins.

Screenshot 2024-05-14 at 11 11 39 AM
Screenshot 2024-05-14 at 11 11 59 AM
Screenshot 2024-05-14 at 11 12 30 AM
Screenshot 2024-05-14 at 11 12 16 AM
Screenshot 2024-05-14 at 11 12 49 AM
Screenshot 2024-05-14 at 11 13 00 AM
Screenshot 2024-05-14 at 11 13 11 AM
Screenshot 2024-05-14 at 11 34 08 AM

## Events

Did you introduce any new tracking events? No
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [x] Have you done sanity checks in the webapp?
- [x] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [x] MobileL (420px)
- [x] Tablet (656px)
- [x] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

MI-146 #done

### Preview domain
https://mi-146-list-view.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/hooks/useFeedLayout.ts
Comment (nensidosari) [inline packages/shared/src/hooks/useFeedLayout.ts:102]: We're only enrolling users that are using list mode

File: packages/extension/src/newtab/CustomLinks.tsx
Comment (nensidosari) [inline packages/extension/src/newtab/CustomLinks.tsx:28]: You'll see these changes a lot - made from this PR because I believe `shouldUseMobileFeedLayout` is not the right term since we're using it for tablet as well.

**Review by kopancek — APPROVED** [review]
Great work, feels really good to remove some of the conditional rendering. It also seems to work as expected in my tests on preview.

Approving, left a non-blocking comment for consideration.

**Review by rebelchris — CHANGES_REQUESTED** [review]
Nice, could only find these two issues E2E testing
And gosh will I be happy if we ever can actually clean all this logic up 😢 

One: (double user bar and top gradient is cut off)
Screenshot 2024-05-13 at 18 17 28
Screenshot 2024-05-13 at 18 17 28

Two: (Section  with search and share seems too small?)
Screenshot 2024-05-13 at 18 18 10
Screenshot 2024-05-13 at 18 18 10

**Comment by nensidosari** [comment]
Thanks @rebelchris. 

> double user bar 

This definitely will fix

> top gradient is cut off)

This actually was [Tsahi approved](https://dailydotdev.slack.com/archives/C0381BQUEN9/p1715264642914339?thread_ts=1715152080.190019&cid=C0381BQUEN9) since we didn't have designs for it - can check again

> Two: (Section with search and share seems too small?)
yup there's some extra padding/margin there will remove

File: packages/shared/src/components/Settings.tsx
Comment (nensidosari) [reply packages/shared/src/components/Settings.tsx:111]: This is ok, there is extra margin without this

File: packages/shared/src/components/feeds/HorizontalFeed.tsx
Comment (nensidosari) [reply packages/shared/src/components/feeds/HorizontalFeed.tsx:107]: Yes 🤦‍♀️

**Review by rebelchris — APPROVED** [review]
More a UX/design concern is that I don't fully understand why the squad header completly changes based on grid/list it should have no impact afaik.
(only on mobile)

But not blocking. (maybe good to raise for design)

**Comment by nensidosari** [comment]
A new issue just showed up with horizontal feed on tablet mode list layout - can't merge until I figure out

**Comment by nensidosari** [comment]
@rebelchris @kopancek apologies for the back and forth on this, I have also fixed the issue of horizontal feed on tablet new list mode + fixed the horizontal feed on current list mode as well.
I know this is approved, and I'll merge after the experimentation details are approved by Ido, but if you want to check again what has changed, please check my last commits.

File: packages/shared/src/components/FeedItemComponent.tsx
Comment (rebelchris) [inline packages/shared/src/components/FeedItemComponent.tsx:199]: Not the cleanest naming i'd say, finding it hard to envision what either does (I know it's mostly legacy, also non blocking :))

File: packages/shared/src/components/feeds/FeedContainer.tsx
Comment (rebelchris) [reply packages/shared/src/components/feeds/FeedContainer.tsx:178]: It's a scrollable feed in some cases though (tag/source page) so not sure the assumption is fully correct around not showing?

File: packages/shared/src/components/squads/SquadPageHeader.tsx
Comment (rebelchris) [inline packages/shared/src/components/squads/SquadPageHeader.tsx:122]: This feels like you could use ternary as well, but up to you?

**Review by rebelchris — APPROVED** [review]
So many conditions on some of this stuff, really hard to follow what can actually impact things, so think we really need to thoroughly check production to avoid accidents.

**Comment by rebelchris** [comment]
Edit, checked the horizontal feed and indeed breaks, I actually think the idea would be that it always renders as cards (not list)?
Screenshot 2024-05-15 at 13 21 25

**Review by rebelchris — APPROVED** [review]

File: packages/shared/src/components/FeedItemComponent.tsx
Comment (nensidosari) [reply packages/shared/src/components/FeedItemComponent.tsx:199]: I know, just wanted to keep the same namings used around at the moment, if this wins we have quite a lot of clean up to do, isList/insaneMode won't be in so many places it is anymore. So just thinking is best to do then.

File: packages/shared/src/components/squads/SquadPageHeader.tsx
Comment (nensidosari) [reply packages/shared/src/components/squads/SquadPageHeader.tsx:122]: hm, it has 2 conditions the allowedToPost and shouldUseListModeV1 which would make the ternary expression a bit messy in my opinion

**Pull request merged**
